policy/lib/sbom/sbom.rego

```diff
@@ -266,3 +266,5 @@ rule_data_attributes_key := "disallowed_attributes"
 rule_data_allowed_external_references_key := "allowed_external_references"
 
 rule_data_disallowed_external_references_key := "disallowed_external_references"
+
+rule_data_allowed_package_sources_key := "allowed_package_sources"
```

Comment:
See the `rule_data_errors` rules above. Let's add a new one that verifies the consistency of this new rule data key. This helps us determine if the rule data provided has any problems.

Reply:
Holding out on this until the the comment about rule data format is settled.

Reply:
Since the rule is specific for cachi2, should it maybe be "cachi2_allowed_sources"?

Reply:
Well, a change in the name of cachi2 is imminent, so I'd like to avoid making this too specific. ([ref](https://github.com/containerbuildsystem/cachi2/discussions/711?sort=old))

What do you think is the right course of action?

Reply:
Rule validation + tests added.